backends/nats/nats.go

```diff
@@ -0,0 +1,486 @@
+package nats
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"errors"
+	"github.com/PowerDNS/simpleblob"
+	"github.com/go-logr/logr"
+	"github.com/nats-io/nats.go"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	// DefaultMaxReconnects - Max Reconnect attempts
+	DefaultMaxReconnects = 5
+	// DefaultReconnectWaitSeconds - Time to wait before attempting reconnect
+	DefaultReconnectWaitSeconds = 1
+	// DefaultNatsBucketDescription - Default description for bucket
+	DefaultNatsBucketDescription = "NATS bucket for simpleblob"
+	// DefaultNatsBucketReplicas - Default replicas for bucket
+	DefaultNatsBucketReplicas = 1
+	// DefaultMaxWaitSeconds - Default max wait for Jetstream replies
+	DefaultMaxWaitSeconds = 10
+)
+
+// internal enum to track auth type
+type authType int
+
+const (
+	username authType = iota
+	token
+	nkey
+	credentials
+	nkeyJwt
+	jwt
+	invalid
+)
+
+// Options enumerates various configuration options
+type Options struct {
+	// Params related to reconnection attempts
+	DisableRetryOnFailedConnect bool `yaml:"disableRetryOnFailedConnect"`
+	MaxReconnects               int  `yaml:"maxReconnects"`
+	ReconnectWaitSeconds        int  `yaml:"reconnectWait"`
+	MaxWaitSeconds              int  `yaml:"maxWaitSeconds"`
+	// Authentication options
+	NatsUsername            string `yaml:"natsUsername"`
+	NatsPassword            string `yaml:"natsPassword"`
+	NatsToken               string `yaml:"natsToken"`
+	NatsNkeySeedFilePath    string `yaml:"natsNkeySeedFilePath"`
+	NatsCredentialsFilePath string `yaml:"natsCredentialsFilePath"`
+	NatsJWTKeyFilePath      string `yaml:"natsJWTKeyFilePath"`
+	// TLS Options
+	// NatsTLSRootCA Specify a custom root CA file (e.g. for self-signed certs)
+	NatsTLSRootCA string `yaml:"natsTLSRootCA"`
+	// NatsTLSClientCert Specify a client cert file
+	NatsTLSClientCert string `yaml:"natsTLSClientCert"`
+	// NatsTLSClientKey Specify a client key file
+	NatsTLSClientKey string `yaml:"natsTLSClientKey"`
```

Comment:
The option names do not clearly indicate that they expect a file path.

Also note the likely live reload requirements, perhaps consider https://github.com/PowerDNS/go-tlsconfig